### ISSUE TITLE
Test extension packages with `pack build --extension` and `pack extension inspect`

### ISSUE THREAD

**Issue by natalieparellano** [buildpacks/pack#1637]
### Description
In order to not block https://github.com/buildpacks/pack/issues/1551 and https://github.com/buildpacks/pack/pull/1616 on extension packages, we won't test them (using images and/or directory extensions instead). This issue is about testing those commands with extension packages.

### Proposed solution
Add unit tests that make use of extension packages

**Comment by nikzayn** [comment]
@natalieparellano I would like to work on this?

**Comment by natalieparellano** [comment]
@nikzayn that would be great! At the moment this may be blocked on https://github.com/buildpacks/pack/issues/1635 as it would be difficult to generate a `.cnb` file to test with without the command existing, but you could try to generate a fixture by hand, it just might be more painful.

**Comment by Parthiba-Hazra** [comment]
Hey, @natalieparellano is this issue already worked on? if not then I would like to work on this, thank you.

**Comment by natalieparellano** [comment]
@nikzayn were you still planning to pick this up? If not @Parthiba-Hazra please go for it :)

**Comment by Parthiba-Hazra** [comment]
Hey @nikzayn are you still working on it?

**Comment by Parthiba-Hazra** [comment]
Hi @natalieparellano could you assign me please? cause @nikzayn is not replying..
And is there anything that you want to share me that might helpful .. as I'm a new folk, thank you..

**Comment by nikzayn** [comment]
Sorry, I was away, yes I am available

**Comment by natalieparellano** [comment]
Awesome @Parthiba-Hazra, I've assigned you.

For testing `pack build --extension` with extension packages I'd take a look at the test for buildpacks here: https://github.com/buildpacks/pack/blob/cc9993c989a263df1c3a0e9020a1367cfbaefc6c/pkg/client/build_test.go#L993

For testing `pack extension inspect` with extension packages I'd look here: https://github.com/buildpacks/pack/blob/cc9993c989a263df1c3a0e9020a1367cfbaefc6c/pkg/client/inspect_buildpack_test.go#L418

**Comment by Parthiba-Hazra** [comment]
Hi @natalieparellano,
sorry for the delay, I was sick.
I write some test cases for **pack build --extension** 
- invalid extension is provided
- multiple extensions are provided
- buildpack and extension both are provided
and I think for **pack extension inspect** necessary test cases already written here
[https://github.com/buildpacks/pack/blob/main/pkg/client/inspect_extension_test.go](url)
is there any other conditions I should also check?

**Comment by natalieparellano** [comment]
@Parthiba-Hazra many apologies, I seem to have forgotten what was the originating need for this issue. You are right that `pack extension inspect` is already covered. For `pack build --extension`, I think the most pressing problem is that there is no `when("Extensions option"` like [exists](https://github.com/buildpacks/pack/blob/cc9993c989a263df1c3a0e9020a1367cfbaefc6c/pkg/client/build_test.go#L686) for buildpacks. Looking at `when("Buildpacks option"`, I think the most relevant contexts from that block are:
* `it("builder order is overwritten"` <- we should use `order for extensions`
* `when("buildpackage image is used"` `it("all buildpacks are added to ephemeral builder"` 
* `when("buildpacks include URIs"` `it("buildpacks are added to ephemeral builder"`
I believe that most of the code paths are shared between buildpacks and extensions and therefore we don't need to duplicate ALL tests for extensions, just the positive paths. Please correct me if you believe otherwise!

**Comment by Parthiba-Hazra** [comment]
Hi @natalieparellano 
The **when("Extensions option"** already exist [here](https://github.com/buildpacks/pack/blob/cc9993c989a263df1c3a0e9020a1367cfbaefc6c/pkg/client/build_test.go#L1713) 
can you check and suggest me any other test cases I should add or not

**Comment by natalieparellano** [comment]
@Parthiba-Hazra thanks for following up here, and apologies for the slow reply. I think we can close this issue as completed :)
